lib/Sat/CryptoMinisat5.cpp

```diff
@@ -183,7 +183,7 @@ uint32_t CryptoMiniSat5::getFixedCountWithAssumptions(const stp::SATSolver::vec_
 
   // The assumptions are each single literals (corresponding to bits) that are true/false. 
   // so in the result they should be all be set 
-  assert(assigned >= assumps.size());
+  assert(static_cast<int>(assigned) >= assumps.size());
```

Comment:
`assumps.size()` returns an `int` and `assigned` is an `unsigned int`; we know "by inspection" that `unsigned int` is "more correct" -- given this, it should be: `assert(assigned >= static_cast<unsigned int>(assumps.size()));`

Reply:
Makes sense. I think that we can assume that `assumps.size()` will never return a negative value.

Reply:
So cast `assumps.size()` to `uint32_t` (and let's hope we never have more than 2^32 assumptions 🙈)

Reply:
Done!